_posts/2020-02-12-myft-machine-learning.html

```diff
@@ -0,0 +1,105 @@
+
+<h2>Follows/unfollows</h2>
+
+{% include figure.html src="/assets/uploads/2020/02/myft_ml_follows_unfollows.png" description="FT concepts positioned according to number of users follows and user unfollows within a selected time frame, points coloured by the type of concept" width="80%" %}
+
+<p>We mapped the number of user concept follows to the number of user concept unfollows to see if we could identify concepts that had a stronger (higher) follow to unfollow ratio count. Armed with this ratio count we could rank the clustered topic suggestions in order of other users voting with their digital feet.</p>
+
+<p>The above graph helped us determine if a topic was more 'sticky', meaning it retains it's gained users more than other concepts.</p>
+
+{% include figure.html src="/assets/uploads/2020/02/myft_ml_follows_unfollows_guides.png" width="80%" %}
+
+<p>The light grey line is x=y, so any concepts on this line have exactly the same number of follows as unfollows. The black line is approximately marking the average 10:1 ratio of follows/unfollows each concept has. The further to the right of the black line a concept is, the greater number of follows it has.</p>
+
+<p>This is far from a perfect ranking system. We discovered that due to some differences in tracking organic follows and organic unfollows we were not able to count organic unfollows with 100% confidence. Leading to concepts appearing on the left of the follow=unfollow grey line, which should not be possible.</p>
+
+<p>We also discovered that while some topics have greater follow ratio, there could be other reasons not to push that to the top of the recommendations pile. Like when was the last time an article was published for this concept? Also there may be acceptable reasons for a large number of unfollows, for example the topic 'World Cup 2018' has seen a reduction in follows because the event has concluded.</p>
+
+<p>This follows/unfollows ratio data gave us the first ranking filter we could use in the proposed recommendation algorithm. Now it was time to tackle our first crack at the FT Labs recommendation engine, now known as Suggestr.</p>
+
+<h2>Suggestr</h2>
+
+<p>We built Suggestr as an endpoint that takes one or more FT topics as input and can suggest other topics of interest. We also included the correlation service, which also powered the correlations game (<a href="https://labs.ft.com/games/2017/09/18/make-connections.html">https://labs.ft.com/games/2017/09/18/make-connections.html</a>), so that we could find topics that appear in the same cluster as the input topic(s), exclude topics that directly correlate and then rank the remaining with their follows/unfollow ratio. This way we suggest topics that don't appear in the same articles as the input topic but have been determined (by ML analysis of user interactions) to be of similar interest.</p>
```

Comment:
Hyper link "correlations game", rather than have the link in brackets. You might want to open in a new tab also, to avoid directing away from this article.

Reply:
Done